lom/src/plugins/sonic/plugin/detection/link/linkcrc/link_crc.go

```diff
@@ -0,0 +1,248 @@
+	outUnicastPacketsMinValue = lomcommon.GetIntConfigurationFromJson(actionKnobsJsonString, out_unicast_packets_min_value_config_key, out_unicast_packets_min_value_default)
+	outlierRollingWindowSize = lomcommon.GetIntConfigurationFromJson(actionKnobsJsonString, outlier_rolling_window_size_config_key, outlier_rolling_window_size_default)
+	minCrcError = lomcommon.GetFloatConfigurationFromJson(actionKnobsJsonString, min_crc_error_config_key, min_crc_error_default)
+	minOutliersForDetection = lomcommon.GetIntConfigurationFromJson(actionKnobsJsonString, min_outliers_for_detection_config_key, min_outliers_for_detection_default)
+	lookBackPeriodInSecs = lomcommon.GetIntConfigurationFromJson(actionKnobsJsonString, look_back_period_in_secs_config_key, look_back_period_in_secs_default)
+
+	// Initialize values.
+	linkCrcDetectionPlugin.counterRepository = &dbclient.CounterRepository{RedisProvider: &dbclient.RedisProvider{}}
+	linkCrcDetectionPlugin.currentMonitoredInterfaces = map[string]LinkCrcDetectorInterface{}
+	linkCrcDetectionPlugin.reportingFreqLimiter = plugins_common.GetDefaultDetectionFrequencyLimiter()
+	err := linkCrcDetectionPlugin.PeriodicDetectionPluginUtil.Init(actionConfig.Name, detectionFreqInSecs, actionConfig, linkCrcDetectionPlugin.executeCrcDetection, linkCrcDetectionPlugin.executeShutdown)
+	if err != nil {
+		lomcommon.LogError(fmt.Sprintf("Plugin initialization failed. (%s), err: (%v)", actionConfig.Name, err))
+		return err
+	}
+	return nil
+}
+
+/* Executes the crc detection logic. isExecutionHealthy is marked false when there is an issue in detecting the anomaly 
+   This is the logic that is periodically executed to detect crc anoamlies */
+func (linkCrcDetectionPlugin *LinkCRCDetectionPlugin) executeCrcDetection(request *lomipc.ActionRequestData, isExecutionHealthy *bool) *lomipc.ActionResponseData {
+	lomcommon.LogInfo(fmt.Sprintf("executeCrcDetection Starting"))
+	ifAnyInterfaceHasCrcError := false
+	var listOfInterfacesWithCrcError strings.Builder
+	currentInterfaceCounters, err := linkCrcDetectionPlugin.counterRepository.GetCountersForActiveInterfaces()
+	if err != nil {
+		/* If redis call fails, there can be no detection that can be performed. Mark it unhealthy */
+		lomcommon.LogError("Error fetching interface counters for LinkCrc detection")
+		*isExecutionHealthy = false
+		return nil
+	}
+
+	if len(currentInterfaceCounters) == 0 {
+		/* currentInterfaceCounters is either nil or 0 which is invalid. Mark it unhealthy */
+		*isExecutionHealthy = false
+		lomcommon.LogError("interface counters is 0")
+		return nil
+	}
+
+        *isExecutionHealthy = true
+	for interfaceName, interfaceCounters := range currentInterfaceCounters {
+		if interfaceCounters == nil {
+			lomcommon.LogError(fmt.Sprintf("Nil interface Counters for %s", interfaceName))
+			*isExecutionHealthy = false
+			continue
+		}
+		linkCrcDetector, ok := linkCrcDetectionPlugin.currentMonitoredInterfaces[interfaceName]
```

Comment:
Say we promise that we scan every N seconds
Say this function, takes M seconds to execute

Is our effective frequency N or N + M ?

Reply:
It is still N.